### ISSUE TITLE
feature_request(toc): Update TOC by save

### ISSUE THREAD

**Issue by Kristinita** [kizza/Table-of-comments#52]
### 1. Summary

It would be nice, if would be possible update TOC by save file.

### 2. Argumentation

Automatic TOC updating. It saves a user time.

### 3. Example

For example, I have `SashaTOC.styl` file.

```stylus
/* ==========================================================================
   > Sasha header
   ========================================================================== */

/* >> Sasha subheader
   ========================================================================== */
```

my TOC:

```stylus
/*
* TOC
*
* Sasha header
* - Sasha subheader
*/
```

I update my headers:

```stylus
/* ==========================================================================
   > Sasha header
   ========================================================================== */

/* >> Sasha first subheader
   ========================================================================== */

/* >> Sasha second subheader
   ========================================================================== */
```

I save `SashaTOC.styl`.

If `update_by_save: true`, It would be nice, if TOC will automatically update:

```stylus
/*
* TOC
*
* Sasha header
* - Sasha first subheader
* - Sasha second subheader
*/
```

### 4. Examples in other apps

[**MarkdownTOC**](https://github.com/naokazuterada/MarkdownTOC) automatically update TOC by save.

Thanks.

**Comment by mihai-vlc** [comment]
This was addressed as part of the 1.2.0 release.

You can enable this behavior from the configuration with the `toc_generate_on_save` option.
Please have a look and let me know if this works on your side.

**Comment by Kristinita** [comment]
@ionutvmi , thanks, it works for me!
